### PR TITLE
DEVEM-521 fix hazel cast metrics  with micrometer

### PR DESCRIPTION
**Pull request by hechmi-dammak-xenit**

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -27,8 +27,8 @@ jobs:
         run: ./gradlew sonarqube
   integration-test:
     runs-on: ubuntu-latest
-    if: ${{ startsWith(github.ref, 'refs/heads/master') || startsWith(github.base_ref, 'master') || startswith(github.ref, 'refs/heads/release') }}
```
Comment (todorinskiz) [inline .github/workflows/ci.yml:30]: Please revert this, failing `master` is enough?

Comment (hechmi-dammak-xenit) [reply .github/workflows/ci.yml:30]: shouldn't the integration test run for all branches before merging?

**Comment by todorinskiz** [comment]
The code looks good to me, but I'm missing unit testing of the:
* Reflection Utils
* Enterprise Cache Metrics

Maybe consider refactoring the ReflectionUtils to a service that we could mock.
This would allow for isolated testing of the reflection utils and EnterpriseCacheMetrics.

**Review by todorinskiz — APPROVED** [review]
LGTM
https://sonarcloud.io/summary/new_code?id=xenit-eu_alfred-telemetry&branch=DEVEM-521

**Review by todorinskiz — APPROVED** [review]

**Pull request merged**